src/adapter/zboss/adapter/zbossAdapter.ts

```diff
@@ -187,124 +210,202 @@ export class ZBOSSAdapter extends Adapter {
     public async changeChannel(newChannel: number): Promise<void> {
-        throw new Error(`Channel change is not supported for 'zboss' yet`);
+        const clusterId = Zdo.ClusterId.NWK_UPDATE_REQUEST;
+        const zdoPayload = Zdo.Buffalo.buildRequest(this.hasZdoMessageOverhead, clusterId, [newChannel], 0xfe, undefined, undefined, undefined);
+
+        await this.sendZdo(ZSpec.BLANK_EUI64, ZSpec.BroadcastAddress.SLEEPY, clusterId, zdoPayload, true /* handled below */);
+        await Wait(12000);
     }
 
     public async setTransmitPower(value: number): Promise<void> {
         if (this.driver.isInitialized()) {
             return await this.queue.execute<void>(async () => {
-                await this.driver.setTXPower(value);
+                await this.driver.execCommand(CommandId.SET_TX_POWER, {txPower: value});
             });
         }
     }
 
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
     public async addInstallCode(ieeeAddress: string, key: Buffer): Promise<void> {
+        logger.error(() => `NOT SUPPORTED: sendZclFrameToGroup(${ieeeAddress},${key.toString('hex')}`, NS);
         throw new Error(`Install code is not supported for 'zboss' yet`);
     }
 
     public async permitJoin(seconds: number, networkAddress: number): Promise<void> {
         if (this.driver.isInitialized()) {
-            return await this.queue.execute<void>(async () => {
-                await this.driver.permitJoin(networkAddress, seconds);
-                if (!networkAddress) {
-                    // send broadcast permit
-                    await this.driver.permitJoin(0xfffc, seconds);
+            const clusterId = Zdo.ClusterId.PERMIT_JOINING_REQUEST;
+
+            if (networkAddress !== undefined) {
+                // TODO: works on networkAddress === 0 for 'coordinator-only'?
+                await this.driver.execCommand(CommandId.ZDO_PERMIT_JOINING_REQ, {nwk: networkAddress, duration: seconds, tcSignificance: 1});
+                // `authentication`: TC significance always 1 (zb specs)
+                const zdoPayload = Zdo.Buffalo.buildRequest(this.hasZdoMessageOverhead, clusterId, seconds, 1, []);
+
+                const result = await this.sendZdo(ZSpec.BLANK_EUI64, networkAddress, clusterId, zdoPayload, false);
+
+                /* istanbul ignore next */
+                if (!Zdo.Buffalo.checkStatus(result)) {
+                    // TODO: will disappear once moved upstream
+                    throw new Zdo.StatusError(result[0]);
                 }
-            });
+            } else {
+                // broadcast permit joining ZDO
+                // `authentication`: TC significance always 1 (zb specs)
+                const zdoPayload = Zdo.Buffalo.buildRequest(this.hasZdoMessageOverhead, clusterId, seconds, 1, []);
+
+                await this.sendZdo(ZSpec.BLANK_EUI64, ZSpec.BroadcastAddress.DEFAULT, clusterId, zdoPayload, true);
```

Comment:
there is a special feature for the ZBOSS - you need to send 2 commands: to the coordinator and the broadcast.
`await this.sendZdo(ZSpec.BLANK_EUI64, networkAddress, clusterId, zdoPayload, true);`

Reply:
So proper logic is like this?

```typescript
if (networkAddress) {
    // device-only when network > 0
    await this.sendZdo(ZSpec.BLANK_EUI64, networkAddress, clusterId, zdoPayload, true);
} else {
    // coordinator-only when networkAddress === 0 or undefined (all)
    await this.sendZdo(ZSpec.BLANK_EUI64, ZSpec.COORDINATOR_ADDRESS, clusterId, zdoPayload, true);

    if (networkAddress === undefined) {
        // all
        await this.sendZdo(ZSpec.BLANK_EUI64, ZSpec.BroadcastAddress.DEFAULT, clusterId, zdoPayload, true);
    }
}

Reply:
yes. or combine in one block with 2 commands sendZdo - always send a broadcast, even if we connect it to some router

Reply:
Not separating it would break what Z2M expects (from the dropdown in frontend).
- `123` => device-only
- `0` => coordinator-only
- `undefined` => all (broadcast)

So I'll push like that, let me know if that doesn't work out in zboss.

Reply:
to join to the coordinator, I have to send 2 commands: one with the coordinator's address (0), and the second with the broadcast. only in this case, the joining of devices begins.
to join to the router, I sent only one command (without broadcast), but the device did not connect to it (the concentrator did not give permission to connect) and I haven't been able to beat it yet. maybe we should send another broadcast team, but I haven't checked yet.

Reply:
I'm seeing a `zb_nlme_permit_joining_request(duration)` in espressif zboss library. Could it be similar to emberznet, where there's a special function to open the network on the coordinator-only without going through a ZDO?